### PR TITLE
Feature/oauth2 jwt as access token

### PR DESCRIPTION
**Pull request by krzpiesiewicz**

### CODE REVIEW

File: engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2Configuration.scala [hunk 1]
```diff
@@ -51,3 +62,37 @@ object ProfileFormat extends Enumeration {
...
+  private case class JwtConfig(publicKey: Option[String],
+                               publicKeyFile: Option[String],
+                               certificate: Option[String],
+                               certificateFile: Option[String]) extends JwtConfiguration {
+    def authServerPublicKey: PublicKey = {
+      val charset: String = "UTF-8"
+
+      def getContent(content: Option[String], file: Option[String]): Option[String] =
+        content.orElse(file map { path =>
+          IOUtils.toString(new FileInputStream(new File(path)), charset)
```
Comment (mproch) [inline engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2Configuration.scala:89]: Source.fromFile(file, StandardCharsets.UTF_8.name()).mkString - nie będziesz musiał importować commons-io

File: engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2Authenticator.scala [hunk 2]
```diff
@@ -27,18 +28,31 @@ class OAuth2Authenticator(configuration: OAuth2Configuration, service: OAuth2Ser
...
+    case OAuth2CollectiveException(errors) => errors.toList.collectFirst { case e@OAuth2ServerError(_) => e }.isEmpty
     case _ => false
   }
 
-  def unapply(t: Throwable): Option[Throwable] = Some(t).filter(apply)
+  trait OAuth2Error {
+    def msg: String
+  }
+
+  case class OAuth2CollectiveException(errors: NonEmptyList[OAuth2Error]) extends Exception {
```
Comment (mproch) [inline engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2Authenticator.scala:47]: Compound or sth like that?

File: engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/DefaultOAuth2ServiceFactory.scala [hunk 3]
```diff
@@ -23,7 +29,24 @@ class DefaultOAuth2Service[ProfileResponse: Decoder](clientApi: OAuth2ClientApi[
...
+
+  private def getProfile(token: String): Future[ProfileResponse] = {
+    val profileFromJwtResult = Validated.fromOption(configuration.jwt.map(new JwtValidator[ProfileResponse](_)),
+      NonEmptyList.of(OAuth2JwtError("authentication.jwt configuration not provided"))) andThen { jwtValidator =>
+      jwtValidator.getProfileFromJwt(token)
+    }
+
+    /* Firstly checks whether a profile can be obtained from the token (provided authentication.jwt configured),
+     * secondly tries to obtain the profile from a sent request.
+     */
```
Comment (mproch) [inline engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/DefaultOAuth2ServiceFactory.scala:42]: I think if the user does not configure jwt, they should not get "authentication.jwt configuration not provided" error?

File: engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2Authenticator.scala (hunk 2 above)
Comment (krzpiesiewicz) [reply engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OAuth2Authenticator.scala:47]: `OAuth2CompoundException`, OK?

**Review by mproch — APPROVED** [review]
LGTM, nice feature :+1:

**Pull request merged**